src/CometWrapper.sol

```diff
@@ -325,23 +323,63 @@ contract CometWrapper is ERC4626, CometHelpers {
     /// @param assets The amount of assets to be converted to shares
     /// @return The total amount of shares computed from the given assets
     function convertToShares(uint256 assets) public view override returns (uint256) {
-        uint64 baseSupplyIndex_ = accruedSupplyIndex();
-        return assets > 0 ? principalValueSupply(baseSupplyIndex_, assets) : 0;
+        return convertToSharesInternal(assets, Rounding.DOWN);
+    }
+
+    /// @notice Allows an on-chain or off-chain user to simulate the effects of their deposit at the current block, given
+    /// current on-chain conditions.
+    /// @param assets The amount of assets to deposit
+    /// @return The total amount of shares that would be minted by the deposit
+    function previewDeposit(uint256 assets) public view override returns (uint256) {
+        return convertToShares(assets);
     }
 
     /// @notice Allows an on-chain or off-chain user to simulate the effects of their mint at the current block, given
     /// current on-chain conditions.
-    /// @param shares The amount of shares to be converted to assets
+    /// @param shares The amount of shares to mint
     /// @return The total amount of assets required to mint the given shares
     function previewMint(uint256 shares) public view override returns (uint256) {
-        return convertToAssets(shares);
+        // Round up so the wrapper does not take a loss
+        return convertToAssetsInternal(shares, Rounding.UP);
     }
 
     /// @notice Allows an on-chain or off-chain user to simulate the effects of their withdrawal at the current block,
     /// given current on-chain conditions.
-    /// @param assets The amount of assets to be converted to shares
+    /// @param assets The amount of assets to withdraw
     /// @return The total amount of shares required to withdraw the given assets
     function previewWithdraw(uint256 assets) public view override returns (uint256) {
-        return convertToShares(assets);
+        // Calculate the quantity of shares to burn by calculating the new principal amount
+        uint64 baseSupplyIndex_ = accruedSupplyIndex();
+        uint256 currentPrincipal = totalSupply;
+        uint256 newBalance = totalAssets() - assets;
+        // Round down so accounting is in the wrapper's favor
+        uint104 newPrincipal = principalValueSupply(baseSupplyIndex_, newBalance, Rounding.DOWN);
+        uint256 shares = currentPrincipal - newPrincipal;
```

Comment:
nit: we can probably just return `currentPrincipal - newPrincipal` instead of assigning to `shares` first

Reply:
Done